tinker_access_client/tinker_access_client/LcdApi.py

```diff
@@ -1,149 +1,229 @@
-                time.sleep(3)
-                self.lcd_byte(line, LCD_CMD)
-                for c in message[i:(len(message))]:
-                    self.lcd_byte(ord(c), LCD_CHR)
-
-        else:
-            messageLJ = message.ljust(LCD_WIDTH, ' ')
-            for i in range(LCD_WIDTH):
-                self.lcd_byte(ord(messageLJ[i]), LCD_CHR)
-
+    # Initialize display
+    def __init__display(self):
+        self.lcd_write(LCD_FUNCTIONSET | LCD_2LINE | LCD_5x8DOTS | LCD_4BITMODE)
+        self.lcd_write(LCD_DISPLAYCONTROL | LCD_DISPLAYON)
+        self.lcd_write(LCD_CLEARDISPLAY)
+        self.lcd_write(LCD_ENTRYMODESET | LCD_ENTRYLEFT)
+        time.sleep(LCD_INITDELAY)
+
+    # Write a command to LCD
+    def lcd_write(self, cmd, mode=0):
+        self.lcd_write_four_bits(mode | (cmd & 0xF0))
+        self.lcd_write_four_bits(mode | ((cmd << 4) & 0xF0))
+
+    # Write four bits to the LCD
+    def lcd_write_four_bits(self, data):
+        self.lcd_device.write_cmd(data | LCD_BACKLIGHT)
+        self.lcd_strobe(data)
+
+    # clocks EN to latch command
+    def lcd_strobe(self, data):
+        self.lcd_device.write_cmd(data | En | LCD_BACKLIGHT)
+        time.sleep(.0005)
+        self.lcd_device.write_cmd(((data & ~En) | LCD_BACKLIGHT))
+        time.sleep(.0001)
+
+    # write a character to lcd (or character rom)
+    # 0x09: backlight | RS=DR< works!
+    def lcd_write_char(self, charvalue, mode=1):
+        self.lcd_write_four_bits(mode | (charvalue & 0xF0))
+        self.lcd_write_four_bits(mode | ((charvalue << 4) & 0xF0))
+
+    # put string function with optional char positioning
+    def lcd_display_string(self, string, line=1, pos=0):
+        if line == 1:
+            pos_new = pos
+        elif line == 2:
+            pos_new = 0x40 + pos
+        elif line == 3:
+            pos_new = 0x14 + pos
+        elif line == 4:
+            pos_new = 0x54 + pos
+
+        self.lcd_write(0x80 + pos_new)
+
+        for char in string:
+            self.lcd_write(ord(char), Rs)
+
+    # clear lcd and set to home
+    def lcd_clear(self):
+        self.lcd_write(LCD_CLEARDISPLAY)
+        self.lcd_write(LCD_RETURNHOME)
+
+    # define backlight on/off (lcd.backlight(1); off= lcd.backlight(0)
+    def backlight(self, state): # for state, 1 = on, 0 = off
+        if state == 1:
+            self.lcd_device.write_cmd(LCD_BACKLIGHT)
+        elif state == 0:
+            self.lcd_device.write_cmd(LCD_NOBACKLIGHT)
+
+    # add custom characters (0 - 7)
+    def lcd_load_custom_chars(self, fontdata):
+        self.lcd_write(0x40);
+        for char in fontdata:
+            for line in char:
+                self.lcd_write_char(line)
+
+    # Write two lines to the LCD, called from outside
     def write(self, first_line, second_line):
-        self.lcd_string(first_line, LCD_LINE_1)
-        self.lcd_string(second_line, LCD_LINE_2)
+
+	# Write the first line
+	for char in first_line:
```

Comment:
Rather than repeating this on line 228, can you make a common function to do this, so that this becomes two lines of code, similar to how it was before? The new function would include the code to move to the correct line.